pages/DSNP/UserData.md

```diff
@@ -0,0 +1,84 @@
+
+If the Operation is successful, any previous data associated with the user for each data type included in the input MUST be removed and replaced by the new data.
+
+Data chunks should be generated for each included data type using the following sequence of operations:
+* Generate the data records, in Avro binary format, according to the versioned schema.
+* Divide the data records into one or more chunks, with each chunk containing a maximum byte count specified by the implementation.
+  * Note that the combined chunks constitute a different, and smaller, binary than a complete Avro file, as they exclude the header information, schema, and any block-level metadata.
+    Each chunk should start and end with a complete Avro object; in other words, chunk boundaries MUST NOT occur mid-record.
+    Applications SHOULD try to include as many records as possible within the byte length limit, but because compression and encryption may alter the byte size versus the raw Avro binary records, they may wish to use a heuristic approach that sometimes produces non-optimal chunks.
+* For each chunk generated, the application should then:
+  * If the data type requires compression, apply the compression codec noted.
+  * If the data type requires encryption,
+    * Retrieve the user's active (most recently announced) `keyAgreement` public key, U<sub>public</sub>.
+    If no key exists, one should be created and published as an Announcement before invoking the Operation.
+    * Create a sealed box, as in the [libsodium](https://doc.libsodium.org/public-key_cryptography/sealed_boxes) function `crypto_box_seal`, using U<sub>public</sub>.
+  * Include the previous `etag` value for the chunk. If the chunk is new, `etag` should be set to `null`.
+  If any chunks are to be deleted, they should be included in the input identified with the existing `etag` and a `null` value for the data.
+
+If the Operation is invoked successfully it MUST (synchronously) return a new set of `etag` values for each data type replaced, corresponding to the updated state of the data (with new chunks added and deleted chunks removed).
+Applications should not interpret this response as an indication that the operation was completed and a state change record emitted, as this typically occurs asynchronously.
+However, this strategy allows applications to make the optimistic assumption that, in due course, the DSNP system will reflect the intended changes, without needing to wait for asynchronous confirmation.
+
+If, on the other hand, an invocation of Replace User Data is rejected due to entity tag discrepancies, this indicates that the relevant data has changed since the last preimage was acquired, and the application should fetch the most current version of the relevant data with the Get User Data Operation, reapply any intended changes, and retry the operation.
```

Comment:
Only use of the word `preimage`. Not sure if it needs additional explanation or use elsewhere?

Reply:
Good callout, will change.